project/SparkBuild.scala

```diff
@@ -436,7 +439,19 @@ object SparkBuild extends PomBuild {
       else x.settings(Seq[Setting[_]](): _*)
     } ++ Seq[Project](OldDeps.project)
   }
+}
 
+object Core {
+  lazy val settings = Seq(
+    resourceGenerators in Compile += Def.task {
+      val buildScript = baseDirectory.value + "/../build/spark-build-info"
+      val targetDir = baseDirectory.value + "/target/extra-resources/"
+      val command =  buildScript + " " + targetDir + " " + version.value
+      Process(command).!!
+      val propsFile = baseDirectory.value / "target" / "extra-resources" / "spark-version-info.properties"
```

Comment:
Maybe a nit: shouldn't all resources be generated into `resourceManaged`? I'm not sure how sbt will handle packaging of non-resourcemanaged artifacts; it could be that spark works locally, however errors may occur in a packaged environment.

I would suggest renaming `"target"/"extra-resources"` to `(resourceManaged in Compile).value`.